### PR TITLE
LPS-198464 and LPS-197723 Update sites MVCResourceCommand with certain info + parameters and OOTB SXP Element features a Site Selector so the user can browse and select the sites to limit the search to without the need look-up for and copy-paste the groupIds (Site IDs)

### PR DESCRIPTION
**Pull request by gustavolimav**
https://liferay.atlassian.net/browse/LPS-198464
https://liferay.atlassian.net/browse/LPS-197723

### CODE REVIEW

**Comment by joshuacords** [comment]
Hi @gustavolimav 

I noticed that GetSitesMVCResourceCommand is copied in both portal-search-tuning-rankings-web and search-experiences-web (we really shouldn't have copied code, rather a Util or something that both could use, but we won't deal with that right now). The only difference was that the [Rankings one](https://github.com/liferay-search/liferay-portal/pull/1202/commits/8282cd86ff1d90a167732c3820da9e2158fe160d#diff-c338be9c4ba72de37a9387ac3d51c12f38b2806619db434a6bc9dcca4d6ac92fR102) uses _addGroupsWithChildren while the other doesn't. I just wanted to make sure this difference was intentional.

File: modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+		Assert.assertEquals(2, sitesJSONObject.getInt("lastPage"));
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 5, sitesJSONArray.length());
+	}
+
+	private Group _createGroup(
+			String descriptiveName, boolean fromCompanyGroupId,
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java:232]: `fromCompanyGroupId` all through this test and in the parallel test needs to be renamed to something like `isActive`.

File: modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+			_getSitesMVCResourceCommand.getSitesJSONObject(_resourceRequest);
+
+		JSONArray sitesJSONArray = sitesJSONObject.getJSONArray("items");
+
+		Assert.assertEquals(
+			sitesJSONArray.toString(), 2, sitesJSONArray.length());
+	}
+
+	@Test
+	public void testGetSitesJSONObjectWithDifferentCompanyGroupId()
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java:137]: `testGetSitesJSONObjectWithDifferentCompanyGroupId` should be renamed `testGetSitesJSONObjectFromNonActiveGroups`

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+				JSONObject fieldJSONObject = fieldsJSONArray.getJSONObject(j);
+
+				fieldJSONObject.put("helpText", "group-ids-help");
+			}
+		}
+
+		return sxpElementJSONObject.toString();
+	}
+
+	private String _replaceHelpTextInElementInstanceJSON(
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java:96]: Let's do better - `_replaceHelpTextForLimitSearchToTheseSites`

**Review by joshuacords — CHANGES_REQUESTED** [review]
After the renamings, everything else looks good.

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java (hunk 3 above)
Comment (gustavolimav) [reply modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java:96]: what about `_replaceHelpTextInElementDefinition`?

**Comment by joshuacords** [comment]
Great changes @gustavolimav. Last thing, _replaceHelpTextInElementInstanceJSON at https://github.com/liferay-search/liferay-portal/pull/1202/commits/e70c1789ad985265223bb4c6b6cf15b9908c74db#diff-d03d0b4721b6d8e4c3fce4e38cb75a49d26654fdf59507253bf5fff7f6c5b397R96 should be the method called `_replaceHelpTextForLimitSearchToTheseSites`, however the other should be renamed too.

Let's use `_replaceElementDefinitionHelpTextForLimitSearchToTheseSites` and `_replaceElementInstanceJSONHelpTextForLimitSearchToTheseSites` for these 2 method names on lines 49 and 96.

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java (hunk 3 above)
Comment (joshuacords) [reply modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v3_1_2/SXPBlueprintUpgradeProcess.java:96]: To me it's important that the name communicates the code is only operating on LimitSearchToTheseSites Element Definitions.

File: modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java (hunk 1 above)
Comment (gustavolimav) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/test/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommandTest.java]: as a note the SF complies about `isActive` the pattern is just `active`

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Pull request closed without merging**